### ISSUE TITLE
Possible Monk Damage Issue

### ISSUE THREAD

**Issue by Mavriked** [EdenServer/community#1406]
I'm not sure what is going on, This is just from me leveling two characters as monk for their first job.

As a Monk my damage has been consistently less than most other DDs. I don't see a difference in damage between being unarmed and having a weapon. Definitely a difference in delay of attack, but with my most recent testing. I've been getting more damage going unhanded and being equipped with a weapon.

Using Deeps I'm doing less damage than the PLD. PLD using a lvl 9 Spatha with a DMG of 11, I'm using Legionnaire's Knuckles DMG +4 Att +3. I also have legionnaire's mittens +3 attack.  Paladin has no attack+ items.  Also, I have maxed h2h combat skill for the lvl.

I noticed this as well earlier while lvling solo. For the most part I avoided low lvl even match mobs, even when i had full HP and TP, since it would end up being too close for comfort.

TL;DR I think that the damage for h1h weapons isn't being calculated, instead the base unarmed attack dmg is what is being done from my experience lvling 2 new characters to lvl 18 on monk.

**Comment by godmode** [comment]
Anyone else having issues with h2h damage?

**Comment by ccl085** [comment]
I did not do test or anything but indeed all mnk I saw (I'm only lvl 50 now) dmg was terrible, so something is most likely wrong as I don't recall mnk being bad in this era

**Comment by Mavriked** [comment]
Forgot to mention, that the 2 characters I lvl'd, one was a Galka and the other was a Mithra and they both had this issue. I would also be willing to do further testing if needed, for example lvling thief using h2h compared to a sword. I remember that h2h was better damage due to combo being such a powerful weapon-skill compared to the available sword weapon-skills. I wasn't planning on lvling Thief for a while, but would be more than willing to do such and contribute to fixing this.

**Comment by beauxq** [comment]
I've paid attention to the base damage calculations (not recently, but in the history of Eden - don't know if anything has changed recently) and I haven't noticed anything wrong.

I've switched h2h weapons and seen a difference in the numbers. So it's not the case that the weapon damage is doing nothing.

What I haven't really paid attention to is comparing it to other jobs / other weapon types, or calculation beyond the weapon damage (except at high pdif, for the crit bug that we've dealt with).

**Comment by Qair** [comment]
<a href="http://tinypic.com?ref=33ngcgh" target="_blank"><img src="http://i65.tinypic.com/33ngcgh.png" border="0" alt="Image and video hosting by TinyPic"></a>

**However** IF you can toss **Asuran Fists** a buff, it would be appreciated

**Comment by Mavriked** [comment]
Looks like it definitely isn't exactly what I was thinking, but I feel like low lvl monk is having damage issues in comparison to other jobs. It was consistently the lowest damage in every PT i joined, that is comparing against thief even prior to sneak attack, rdm, and pld. Warriors and Sams were doing double the monk damage, which I'm not as against. I do believe warrior and sams should be doing more damage, but it feels so off the damage that I was doing. Even with Att+ items, Courage Ring, and food buffs. Again my experience is only in leveling from 1-18 on two characters. Higher lvls, I have no idea if monk starts evening out with other damage dealers.

**Comment by Qair** [comment]
monk is not too great at high level either, it probably loses to every equally geared job that isn't thief. Difference in weaponskill damage is hard to compensate for.

**Comment by Qair** [comment]
Even at higher levels mnk suffers as you're usually not able to sub war in exp parties because of mp sink. As /nin without brutal earring, it can be pretty tough and I'm not totally convinced asuran fists is where it's suppose to be at as far as damage is concerned. In my last merit pt, my mnk was average 450 damage on asuran fists, while other dps where doing about 750 damage on average. So if you're equally geared, you can't really make up the difference in weapon-skill damage.

**Comment by Mavriked** [comment]
I'm going to just report on partys that i have with a monk. Included is a snippet of Deeps with a war/mnk w/ GA, war/nothing w/ GA, war/nothing w/axe&shield, and Lakshmi a Mnk/War. As can be seen Lakshmi is significantly below us.
![Monk1](https://user-images.githubusercontent.com/48971707/55207069-cf8c2500-51a6-11e9-9f80-e87be32bd1ab.PNG)

**Comment by Serephim** [comment]
Monk is just extremely underwhelming as a DPS on this server. I have it geared the hell out in amazing gear with 22% equipment haste and it still does way less DPS than all of my other DD classes. It doesn't feel retail at all. Black Belt Monk was really good on retail. Here, it's just not very good at all. Out classed by nearly every damage dealing job. Even my BST out DPS's it. And that's just pathetic.

**Comment by VerrenGunnar** [comment]
I dont know if its exactly a "bug", but yeah it feels like something about the dmg calc on the server vs the retail one leaves HtH sorely lacking, i meen all it takes is a 30 look at the discord to see that the prevailing view is that "Monk is trash"

**Comment by VerrenGunnar** [comment]
https://genomeffxi.livejournal.com/2007/01/18/
just for consideration, an in era blog that talks a lot about all things monk, shows mnk dmg in era being a lot better then it is currently on eden

**Comment by gweivyth** [comment]
Just a heads up and so that I can link all the tickets together, me, Qair, and Setzor spent a couple of hours messing around with H2H weapon skills (#1450 and #1485) and found that they are not properly accounting for your H2H skill.  If you look [here in DSP](https://github.com/DarkstarProject/darkstar/blob/3ed7b224e1bfcd71e6ae335fbc190f353c9e5d3a/scripts/globals/weaponskills.lua#L40) you'll see what I mean.  Setzor said he sort of just poked around and stumbled into it lol. 

Looks like they're using 11%+3 of your H2H skill, which seemed like it might have just been an arbitrary guess when it was implemented on DSP.  We live-tested a bunch of different variables in the place of that 11% and landed on a number that is much more reasonable, at least at 75.  

I don't remember the exact number Setzor said we were on when we determined that we had probably found the "sweet spot," but I believe it was 20%~.  At this point we were averaging much more reasonable Asuran Fists, with Riaq in mid-tier gear parsing around 630~ Asurans with an average of 6 hits landing, and spiking at 1k~ when all 8 hit.  This was consistent with the source material that @VerrenGunnar provided in the previous post.  Obviously this sample size wasn't gigantic, (about 100 WS') but it seemed like a reasonable place to adjust to, and can easily be tweaked later by just changing one variable.  All of this testing was done on Greater Colibri, as this was the baseline that we wanted to use for these tests since they're neutral to blunt damage, are the appropriate level, and are a common mob that people merit on.

It is still % based though so it might not have as big of an impact at lower levels, but you should be able to see the power creep as you level up.  Monk (and by proxy PUP) should see a pretty decent bump in consistency and damage in their weapon skills.

Thanks to @Setz0r and @Qair for working on this.  It was a fun evening in voice chat throwing hands.  <3

**Comment by isotor** [comment]
Interesting find, but just to chip in here, the `(skill *0.11) + 3` seems to be a very established figure on fan sites:

- https://www.bg-wiki.com/bg/Category:Hand-to-Hand
- https://ffxiclopedia.fandom.com/wiki/Category:Hand-to-Hand

Given it's literally just the base damage of swinging bare-fisted, I can't see any reason why this might be a value that wouldn't have been absolutely nailed down over the game's long history. 

I'm not saying you're wrong, and hell if it makes the job suck less even if it's a bandaid an improvement is an improvement - just maybe this isn't the droid you're looking for.

**Comment by isotor** [comment]
Also, for reference: https://github.com/DarkstarProject/darkstar/blob/3ed7b224e1bfcd71e6ae335fbc190f353c9e5d3a/src/map/entities/battleentity.cpp#L1487

**Comment by gweivyth** [comment]
> Interesting find, but just to chip in here, the `(skill *0.11) + 3` seems to be a very established figure on fan sites:
> 
> * https://www.bg-wiki.com/bg/Category:Hand-to-Hand
> * https://ffxiclopedia.fandom.com/wiki/Category:Hand-to-Hand
> 
> Given it's literally just the base damage of swinging bare-fisted, I can't see any reason why this might be a value that was never absolutely nailed down over the course of the game's long history.
> 
> I'm not saying you're wrong, and hell if it makes the job suck less even if it's a bandaid an improvement is an improvement - just maybe this isn't the droid you're looking for.

Yeah I noticed that on the wiki as well but I'm not entirely sure where they even derived that number from, especially from a strictly player perspective.  It is oddly specific given how obfuscated FFXI's numbers always were back then.

**Comment by Qair** [comment]
I'm going to put on my conspiracy theory hat here for a second. What if the weaponskill formula for h2h on dps isn't taking the weapon damage into consideration or if it is, it's too little.

The formula according to that wiki is skill x .11 + 3, so a mnk at 75 would have a base damage of 
276 x.11 +3 = 33 (rounded down) 
33 (base damage) + 18 (weapon damage) so with destroyers it comes out to be 33+18 = 51 total damage

In testing we used a modifier of 276 x  .2 +3 = 58

If the weaponskill formula isn't including the 18 dmg from destoryers, then the 58 base damage from h2h is fairly close to the base damage + weapon damage at .11. Meaning, is it possible that during weaponskills is it only calculating 33 base damage?

**Comment by VerrenGunnar** [comment]
well in general i dont think its properly taking weapon dmg in for hth attacks, looking at the above screenshot, while i dont know what weapon they used, 85 to110 feels like too small of a dmg jump from no weapon to a weapon

**Comment by Qair** [comment]
I'm not really sure about the h2h damage being too low with a weapon, but I do think the tp return seems to be too low. I've seen 18 tp return asuran fists in era and the max tp return you can get on eden is 158 which seems odd.

**Comment by Aneyus** [comment]
Those 18 TP returns possibly used usukane sune-ate and/or rajas ring for more Store TP as well. Can't just eyeball without all the info

**Comment by MGramolini** [comment]
Is 18 TP supposed to be a good return? that wasn't a typo?

**Comment by Qair** [comment]
in retail, before the change from 100 to 1000 tp, 16 tp return on asuran seems like it was the max return. so 18 was a full return + store pt @MGramolini. I'm not sure how the conversion works on here, or why it's 158 instead of 160 tp return.

**Comment by gweivyth** [comment]
Looks like it might just be not rounding properly. It should be 160, yeah. People seem to believe that Asuran could double attack but I think they're just under geared and getting random 8-hits while wearing their Raja's.

**Comment by Qair** [comment]
There's a lot of things I noticed:

1.) The max tp return should be 160
2.) Out of all the asuran fists video's ive watched, I've never seen it go below 10 tp, it always falls in the 10-16 range. I'm not sure how that's converted in the 1000 tp scale or why you can get under 100 on Eden .
3.) I fought 4 VT imps with and without destroyers last night
Bare handed melee range was =  38-54; WSavg = 432 (148-154 tp return) Bare handed only gets 154 tp return max.
Destroyer melee range was = 43-74 : WS avg = 540 (148-158 tp return) 

if I had included non 148-154//158 tp returns, the average would be in the 300-400's

I tested it on VT imps because I wanted to comapare it to this video:
[https://www.youtube.com/watch?v=riV1Q20bADE](url)

He has high's of 80's but I don't know food he was using, his asuran average was also about 650 average

**Comment by gweivyth** [comment]
The people in that thread I like linked talked about tp returns differing between weapons. When Setzor and I did our TP tests I'm 99% certain I was wearing Cross-Counters so my returns might have been incorrect. I would have to go back and check.

**Comment by Qair** [comment]
The people in the videos i linked are using cross counters.

Example of a mnk with walmart gear doing 500+ dmg howling fists in retail era, can't barely break 350 on eden.
https://www.youtube.com/watch?v=8pIdCc20c-k

**Comment by MGramolini** [comment]
Anything between 156 and 164 would probably show as 16 back in the day on retail. It depends on how their rounding worked. We know how the math on TP returns works in general so we just need to make sure it is applied correctly.

If you are getting 158 I would bet that it is correct with current gear values.

Do you get 88-158 and occasionally a number between 10-70 which is always divisible by 10?

**Comment by gweivyth** [comment]
I've gotten lower than 88. I was seeing 70s when I whiffed one of my two initial hits. (Asuran being technically "both fists" + 6 additional hits) Setzor has a script that will tell you how many hits you land every time you weapon skill plus your TP return.

**Comment by MGramolini** [comment]
Oh ok that’s right so the examples I gave would need some tweaking. Most likely something along the lines of 49 tp per fist initial hit and 10 for each additional.

**Comment by MGramolini** [comment]
I think based on some code I have seen in DSP that they could be incorrectly subtracting a constant in the WS calculation.

**Comment by Qair** [comment]
I'll repost this in this thread but i parsed riaq and in 53 weaponskills this is what I came up with 

Max TP possible for Asuran Fists = 158 tp, which seems low but that's besides the point
Riaq's melee Accuracy was 94.43%
Riaq's average weaponskill return was 123 tp per asuran
Riaq's weaponskill average was 412 damage with MINUET taking my attack from 394 to 451.

I'm not sure how multihit weaponskills are caluclated so I just divided 158 / 8 to get 19.75 tp per hit.

Conclusion -  5.9 hits per weaponskill
in 53 weaponskill attempts, I only had 4 return 158 tp. 4/53 = 7.5% chance to land all 8 hits in this sample.

**Comment by Qair** [comment]
Parsed myself in Dynamis, so mostly level 77-79 damage mobs

Average tp return per asuran fists = 126.7091
Average asuran fists damage = 619 (most were activated with mithkabob)

Raw TP numbers:

148
--
89
158
128
148
138
89
158
79
148
89
158
99
99
158
148
89
99
138
128
148
158
138
99
128
148
128
158
138
69
158
138
158
99
128
158
138
158
148
89
108
148
148
89
158
99
40
158
79
138
148
99
99
148
138

**Comment by MGramolini** [comment]
7 5 8 5 7 6 5 8 4 7 5 8...

Etc. hits per WS starting from the top 

69 and 108 being the only 2 WS with less than 4 hits (3).

**Comment by Qair** [comment]
Accuracy sucks but that being said there's something fundamentally wrong with the ws formula for h2h. Something isn't being modified correctly.

**Comment by MGramolini** [comment]
I don't think its just h2h.  Tachi Gekko is weaker than it should be as far as i can tell as well.

**Comment by Qair** [comment]
I says h2h because it seems to effect all h2h weaponskills where as gk seems pretty well off for the most part outside of gekko. I can't imagine it's a multihit weaponskill problem either as rampage seems fine but dancing edge is also shit.

**Comment by godmode** [comment]
How is auto attack damage?

**Comment by Qair** [comment]
Auto attack range is pretty close from what I can tell from videos of era. Weaponskills are missing alot of damage which makes monk useless for everything. Pup will also have this problem. In order to average 600-700 damage asuran fists vs colibri, you would need a h2h with around 45ish damage on it. Destroyers have 18. Impossible to reach.

**Comment by godmode** [comment]
Would you say weapon skill damage is about half of what it should be?

**Comment by Qair** [comment]
Feels that way. In testing with setzor raisingtje h2h skill formula from h2h skill * 11 + 3 to h2h skill * 2.0 + 3 gave about a 640ish asyran fist average vs greater colib but it is just a bandaid fix. There's a multiplier problem

**Comment by gweivyth** [comment]
Yeah towards the beginning of the thread I showed the line in DSP where WS' for Monk were pulling 11% of skill.  When we adjusted to 20% it seemed much more in-line.  The accuracy felt a little high with that modifier (a lot of 158 returns) but the actual raw damage didn't seem crazy high, (the proper average of about 650) but rather just where it should be.  So this definitely seems like a damage calculation issue. 

I mean if we're okay with this being the band-aid fix it doesn't seem like a bad one to me, the TP returns being high isn't gamebreaking or anything, but if someone wanted to dig for the root of the problem we definitely haven't managed to uncover that yet.

**Comment by isotor** [comment]
Are you guys sure this isn't an issue with other accuracy-based multihit WS like Penta and Dancing Edge?

**Comment by Qair** [comment]
There is an issue with "acc varies with tp" weapon skills,. Initially i thought Asuran was bad becasue of the acc penalty to such weaponskills, but when we set it to 1 instead of .8 on the test server Asuran was still only averaging about 450 damage vs greater colib. There's a lot of damage disappearing something. I can post videos of mnks do 500+ dmg howling fists in era vs greater colib while my mnk does like 250-300 dmg. Asuran averages around 350-400 without buffs and around 450 damage with a corsair vs greater colibs. Raging fists isn't really any better.

**Comment by isotor** [comment]
The above doesn't answer the question of whether other accuracy-based multihit WS are also still under-performing.

**Comment by gweivyth** [comment]
> The above doesn't answer the question of whether other accuracy-based multihit WS are also still under-performing.

It doesn't seem consistent across all multihits.  Rampage, Vorpal Blade, and Penta Thrust all seem fine.  Dancing Edge and Asuran Fists do not.  I can't speak to any of the other ones since I only have WAR/MNK/DRG/PLD/THF leveled.

**Comment by isotor** [comment]
Fishing here a bit, as can't see any reason penta would be fine and Asuran/DE not - could is perhaps be that people are generally more likely to have Polearm merited than Dagger/H2H?

**Comment by gweivyth** [comment]
> Fishing here a bit, as can't see any reason Penta would be fine and Asuran/DE not; is it possible that players are generally more likely to have polearm merited than dagger/h2h when making these comparisons?

I don't think so.  I don't have any merits in any of the weapons I listed except for Sword, but even before I had those Vorpal was pretty on-par.

**Comment by Qair** [comment]
Dancing edge and Asuran fists are both underperforming. I think penta might be a little too inaccurate as a .8 acc modifier is fairly harsh for non sidewinder weaponskills. It's hard for me to know the extent of penta thrust as nobody goes to endgame as drg in my linkshell and we don't merit on anything that isn't birds so they always have a piercing bonus. Fortunately for thf, it has sharkbite, where as h2h classes have no good secondary weaponskill. They're all just trash.

**Comment by MGramolini** [comment]
Didn’t people used to use dragon kick in some high level scenarios?  As for testing....

We should make sure we always test accuracy with toooo much so we cap at 95%. With 95% and 1000 weaponskills we should be able to see if the .8 modifier is correct. Maybe it should be .9 and .95?

**Comment by Qair** [comment]
there's no evidence that asuran fists ever had a .8 modifier. Even in patch notes, the only things that had a negative accuracy modifier removed from them was penta thrust and guillotine.

**Comment by Qair** [comment]
> 
> 
> Yeah towards the beginning of the thread I showed the line in DSP where WS' for Monk were pulling 11% of skill. When we adjusted to 20% it seemed much more in-line. The accuracy felt a little high with that modifier (a lot of 158 returns) but the actual raw damage didn't seem crazy high, (the proper average of about 650) but rather just where it should be. So this definitely seems like a damage calculation issue.
> 
> I mean if we're okay with this being the band-aid fix it doesn't seem like a bad one to me, the TP returns being high isn't gamebreaking or anything, but if someone wanted to dig for the root of the problem we definitely haven't managed to uncover that yet.

Keep in mind that the 2.0 fix also had the 1.0 acc modifier, so if 2.0 is implemented on the server, assuran will still have a .8 acc modifier which will make the damage lower.

**Comment by Caelic** [comment]
> xizeta self-assigned this 8 hours ago
Bless your soul, kind stranger.

**Comment by xizeta** [comment]
This has not been forgotten. We've been collecting retail data and this is being investigated.

**Comment by xizeta** [comment]
I've collected hours and hours of data, at different levels, with different weapons and even different jobs to rule out any source of differences. In conclusion, the hit for hit melee damage of Hand-to-Hand on Eden is not lower, and it scales with weapon DMG the same way retail does.

Regarding the earlier mentioned Paladin damage comparison. Hand-to-Hand weapons deal sustained damage, at a fast rate. The hit for hit damage should not be equal to other melee weapons such as swords, especially Spatha which is a Long Sword, meaning it has particularly high DMG.

There is definitely something wrong with Asuran Fists and possibly other Hand-to-Hand weapon skills. However, this is already covered by other tickets and they are being investigated.

**Comment by bkesecker** [comment]
After new login server work, I'll start looking into weaponskill damage for MNK.

**Comment by Qair** [comment]
Is there any progress being made on this?

**Comment by Mozzy4Ever** [comment]
I believe this bug report should have been closed with the update that fixed multiple weaponskills (mainly ones where accuracy varied with TP). I haven't really heard many complaints about MNK damage since that update. #1485 being one of the reports that was closed related to this

**Comment by Garphilius** [comment]
This should definitely be closed. MNK dmg was fixed long ago. Asuran Fists does era-like DMG now.
